packages/dds/tree/src/shared-tree/treeAlpha.ts

```diff
@@ -481,6 +513,72 @@ export const TreeAlpha: TreeAlpha = {
 		const parentSchema = treeNodeApi.schema(parent);
 		return getPropertyKeyFromStoredKey(parentSchema, storedKey);
 	},
+
+	child: (node: TreeNode, key: string | number): TreeNode | TreeLeafValue | undefined => {
+		const flexNode = getOrCreateInnerNode(node);
+		debugAssert(() => !flexNode.context.isDisposed() || "FlexTreeNode is disposed");
+
+		const schema = treeNodeApi.schema(node);
+		const storedKey = tryGetStoredKeyFromPropertyKey(schema, key);
+		if (storedKey === undefined) {
+			return undefined;
+		}
+
+		if (isArrayNodeSchema(schema)) {
+			const sequence = flexNode.getBoxed(EmptyKey);
+
+			const index =
+				typeof storedKey === "number"
+					? storedKey
+					: asIndex(storedKey, Number.POSITIVE_INFINITY);
+
+			if (index === undefined) {
+				return undefined;
+			}
+
+			const child = sequence.boxedAt(index);
+			if (child === undefined) {
+				return undefined;
+			}
+
+			return getOrCreateNodeFromInnerNode(child);
+		}
+
+		const field = flexNode.tryGetField(brand(String(storedKey)));
+		if (field !== undefined) {
+			return getTreeNodeForField(field);
+		}
+
+		return undefined;
+	},
+
+	children: (node: TreeNode): [string | number, TreeNode | TreeLeafValue][] => {
+		const flexNode = getOrCreateInnerNode(node);
+		debugAssert(() => !flexNode.context.isDisposed() || "FlexTreeNode is disposed");
+
+		const schema = treeNodeApi.schema(node);
+
+		const result: [string | number, TreeNode | TreeLeafValue][] = [];
+
+		if (isArrayNodeSchema(schema)) {
+			const sequence = flexNode.getBoxed(EmptyKey);
+
+			for (const childFlexNode of sequence.boxedIterator()) {
+				const childTreeNode = getOrCreateNodeFromInnerNode(childFlexNode);
+				result.push([childFlexNode.parentField.index, childTreeNode]);
+			}
+		} else {
+			for (const childFlexField of flexNode.boxedIterator()) {
```

Comment:
Using FlexTreeNode.keys to loop over then tryGetField and passing the result into getTreeNodeForField will avoid allocating unnecessary flex tree nodes and fields (which is pretty much all the fields and all leaf nodes: it could save a lot of overhead, which are all live until disposed, so could use memory until app lifetime is over)